_layouts/home.html

```diff
@@ -23,6 +23,17 @@ <h3>
         </h3>
         {%- if site.show_excerpts -%}
           {{ post.excerpt }}
+          {% capture content_words %}
+            {{ post.content | number_of_words }}
+          {% endcapture %}
+
+          {% capture excerpt_words %}
+            {{ post.excerpt | number_of_words }}
+          {% endcapture %}
+
+          {% if content_words != excerpt_words  %}
+            <p><a href="{{site.baseurl}}{{ post.url }}">Read more</a></p>
```

Comment:
Please prefer using Jekyll's `URLFilters` when URLs come into the picture:
```
<p><a href="{{ post.url  | relative_url }}">Read more</a></p>
```

Reply:
Updated the PR.